File: core/java/src/main/java/hara/lang/protocol/IEncode.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iencode", name = "IEncode")
public interface IEncode {
  @HaraMethod(value = "encode", arity = 2)
  Object encode(Object options);
}

File: core/rust/src/lang/protocol/iencode.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iencode", name = "IEncode")]
pub trait IEncode<O> {
    type Output;

    #[hara_method(value = "encode", arity = 2)]
    fn encode(&self, options: O) -> Self::Output;
}
